packages/x-data-grid/src/models/gridSlotsComponentsProps.ts

```diff
@@ -35,6 +35,8 @@ import type { GridRowCountProps } from '../components/GridRowCount';
 import type { GridColumnHeaderSortIconProps } from '../components/columnHeaders/GridColumnHeaderSortIcon';
 
 type DividerProps = {};
+type RootProps = Record<string, unknown>;
+type MainProps = Record<string, unknown>;
```

Comment:
It should rather be:

```suggestion
type RootProps = React.HTMLAttributes<HTMLDivElement>;
type MainProps = React.HTMLAttributes<HTMLDivElement>;
```

Reply:
It caused this issue:
![image](https://github.com/user-attachments/assets/eb0f03d2-7235-4f54-9b32-f4f6fe42c46e)

I assume the same was the reason we didn't use `React.HTMLAttributes<HTMLDivElement>` in the initial [PR](https://github.com/mui/mui-x/pull/8845) when supporting forwarded props.

@romgrk Could you recall by any chance if it was the same reason?

Reply:
Yeah, `React.HTMLAttributes` does not cover data attributes: https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/69864

We can do this instead:
```ts
React.HTMLAttributes<HTMLDivElement> & Record<`data-${string}`, string>
```